src/coprocessor/dag/expr/compare.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+use std::cmp::Ordering;
+
+use tipb::expression::ScalarFuncSig;
+
+use coprocessor::codec::{datum, mysql, Datum};
+use super::{StatementContext, Result, FnCall};
+
+impl FnCall {
+    pub fn compare_int(&self,
+                       ctx: &StatementContext,
+                       row: &[Datum],
+                       sig: ScalarFuncSig)
+                       -> Result<Option<i64>> {
+        let lhs = try!(self.children[0].eval_int(ctx, row));
+        let rhs = try!(self.children[1].eval_int(ctx, row));
+        match (lhs, rhs) {
+            (None, None) if sig == ScalarFuncSig::NullEQInt => Ok(Some(1)),
+            (Some(lhs), Some(rhs)) => {
+                let lhs_unsigned = mysql::has_unsigned_flag(self.children[0].get_tp().get_flag());
+                let rhs_unsigned = mysql::has_unsigned_flag(self.children[1].get_tp().get_flag());
+                let ordering = cmp_i64_with_unsigned_flag(lhs, lhs_unsigned, rhs, rhs_unsigned);
+                Ok(Some(match sig {
+                    ScalarFuncSig::LTInt => ordering == Ordering::Less,
+                    ScalarFuncSig::LEInt => ordering != Ordering::Greater,
+                    ScalarFuncSig::GTInt => ordering == Ordering::Greater,
+                    ScalarFuncSig::GEInt => ordering != Ordering::Less,
+                    ScalarFuncSig::NEInt => ordering != Ordering::Equal,
+                    ScalarFuncSig::EQInt | ScalarFuncSig::NullEQInt => ordering == Ordering::Equal,
+                    _ => unreachable!(),
+                } as i64))
+            }
+            _ => Ok(None),
```

Comment:
if `sig` is `NullEq`, and `lhs` is `None`, but `rhs` is `Some`, should return `Ok(false)`.